src/api/app/models/project_log_entry.rb

```diff
@@ -27,9 +27,8 @@ def self.create_from(payload, created_at, event_model_name)
     entry
   end
 
-  # Delete old entries
-  def self.clean_older_than(date)
-    where('datetime < ?', date).delete_all
+  def self.cleanup
+    where(event_type: [:build_fail, :build_success]).where('datetime < ?', Time.zone.yesterday.beginning_of_day).find_each(&:delete)
```

Comment:
why not delete_all? There can be a lot of them after all

Reply:
From build.o.o

```ruby
ProjectLogEntry.where(event_type: [:build_fail, :build_success]).where('datetime > ?', Time.zone.yesterday.beginning_of_day).count
=> 45580
```

And we only started yesterday at 20:00...

Reply:
How does this answer my question? delete_all is one SQL statement - your code is 45581 SQL statements.

Reply:
Fear of locks :cowboy_hat_face: 

https://www.youtube.com/watch?v=RwpQuvNP0R8

Reply:
The only problem with `delete_all` is that it goes straight to the database skipping things like callbacks and dependent rules, but we don't have anything like that in those models :+1:

Reply:
Yeah so does `.delete`. I was worried about it taking too long but I guess this was unfounded.